src/algorithms/calorimetry/CalorimeterClusterRecoCoG.cc

```diff
@@ -161,3 +162,162 @@ void CalorimeterClusterRecoCoG::AlgorithmProcess() {
+  cl.setIntrinsicPhi(edm4eic::angleAzimuthal(cl.getPosition()));
+  // TODO errors
+
+  //_______________________________________
+  // Calculate cluster profile:
+  //    radius,
+  //   	dispersion (energy weighted radius),
+  //   	sigma_long
+  //   	sigma_short
+  //   	sigma_z
+  double radius = 0, dispersion = 0, lambda_1 = 0, lambda_2 = 0, lambda_3 = 0;
+  double w_sum = 0;
+  double sum_11 = 0, sum_22 = 0, sum_33 = 0;
+  double sum_12 = 0, sum_13 = 0, sum_23 = 0;
+  double sum_1 = 0, sum_2 = 0, sum_3 = 0;
+
+  if (cl.getNhits() > 1) {
+
+    for (const auto& hit : pcl->getHits()) {
+
+      const auto delta = cl.getPosition() - hit.getPosition();
+      radius += delta * delta;
+
+      double w = weightFunc(hit.getEnergy(), cl.getEnergy(), m_logWeightBase, 0);
+      w_sum += w;
+      dispersion += delta * delta * w;
+
+      double pos_1 = edm4eic::anglePolar( hit.getPosition() );
+      double pos_2 = edm4eic::angleAzimuthal( hit.getPosition() );
+      double pos_3 = hit.getPosition().z;
+
+      if( m_xyClusterProfiling ) {
+        pos_1 = hit.getPosition().x;
+        pos_2 = hit.getPosition().y;
+      }
+      sum_11 += w * pos_1 * pos_1;
+      sum_22 += w * pos_2 * pos_2;
+      sum_33 += w * pos_3 * pos_3;
+      sum_12 += w * pos_1 * pos_2;
+      sum_13 += w * pos_1 * pos_3;
+      sum_23 += w * pos_2 * pos_3;
+      sum_1  += w * pos_1;
+      sum_2  += w * pos_2;
+      sum_3  += w * pos_3;
+    }
+
+    if( w_sum > 0 ) {
+      radius = sqrt((1. / (cl.getNhits() - 1.)) * radius);
+      dispersion = sqrt( dispersion / w_sum );
+
+      // variances and covariances
+      double sigma_11 = sum_11 / w_sum - (sum_1/w_sum) * (sum_1/w_sum);
+      double sigma_22 = sum_22 / w_sum - (sum_2/w_sum) * (sum_2/w_sum);
+      double sigma_33 = sum_33 / w_sum - (sum_3/w_sum) * (sum_3/w_sum);
+      double sigma_12 = sum_12 / w_sum - (sum_1/w_sum) * (sum_2/w_sum);
+      double sigma_13 = sum_13 / w_sum - (sum_1/w_sum) * (sum_3/w_sum);
+      double sigma_23 = sum_23 / w_sum - (sum_2/w_sum) * (sum_3/w_sum);
+
+      // covariance matrix
+      Eigen::MatrixXd cov3(3,3);
+      cov3(0,0) = sigma_11;  cov3(1,1) = sigma_22;  cov3(2,2) = sigma_33;
+      cov3(0,1) = sigma_12;  cov3(0,2) = sigma_13;  cov3(1,2) = sigma_23;
+      cov3(1,0) = cov3(0,1); cov3(2,0) = cov3(0,2); cov3(2,1) = cov3(1,2);
+
+      // Eigenvalues correspond to cluster's 2nd moments (sigma_long, sigma_short, sigma_z)
+      Eigen::EigenSolver<Eigen::MatrixXd> es(cov3, false); // set to true for eigenvector calculation
+      auto eigenValues = es.eigenvalues();
+      lambda_1 = eigenValues[0].real(); // imaginary parts correspond to unphysical roots
+      lambda_2 = eigenValues[1].real();
+      lambda_3 = eigenValues[2].real();
+    }
+  }
+
+  cl.addToShapeParameters( radius );
+  cl.addToShapeParameters( dispersion );
+  cl.addToShapeParameters( lambda_1 );
+  cl.addToShapeParameters( lambda_2 );
+  cl.addToShapeParameters( lambda_3 );
```

Comment:
We need to document the meaning and units in
https://github.com/eic/EDM4eic/blob/c8ffba77708dfd01a9f1ee2134001d59e47920cb/edm4eic.yaml#L249

Reply:
It's mentioned that one should alter the "metadata".  How do I do that?

Reply:
I don't think it's possible to do that now: AIDASoft/podio#407
For now, just updating the comment in EDM4eic repo would suffice.